low-level/files/opt/seagate/sspl/setup/sspl_prepare.py

```diff
@@ -62,20 +62,10 @@ def validate(self):
                 "server_node>%s>bmc>ip" % machine_id)
             bmc_user = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
                 "server_node>%s>bmc>user" % machine_id)
-            secret = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+            bmc_secret = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
                 "server_node>%s>bmc>secret" % machine_id)
-            key = Cipher.generate_key(cluster_id, "cluster")
-            bmc_passwd = Cipher.decrypt(key, secret.encode("ascii")).decode()
-            enclosure_id = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-                "server_node>%s>storage>enclosure_id" % machine_id)
-            primary_ip = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-                "storage_enclosure>%s>storage>controller>primary>ip" % enclosure_id)
-            secondary_ip = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-                "storage_enclosure>%s>storage>controller>secondary>ip" % enclosure_id)
-            cntrlr_user = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-                "storage_enclosure>%s>storage>controller>user" % enclosure_id)
-            cntrlr_passwd = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-                "storage_enclosure>%s>storage>controller>password" % enclosure_id)
+            bmc_key = Cipher.generate_key(cluster_id, ServiceTypes.SERVER_NODE.value)
+            bmc_passwd = Cipher.decrypt(bmc_key, bmc_secret.encode("utf-8")).decode("utf-8")
```

Comment:
how bmc_passwd different from bmc_secret? decryption n use of secret looks proper

Reply:
Please find the answer in above reply.